backend/src/layer/data/flickr_api/flickr_api_handler.rs

```diff
@@ -18,35 +18,74 @@ pub struct FlickrApiHandler {
 
 lazy_static! {
     static ref LONG_URL_REGEX: Regex =
-        Regex::new(r"(https://www.flickr.com/photos/)(\w+)/(\d+)([/]{0,1})")
+        Regex::new(r"(https://www\.flickr\.com/photos/)([\w@]+)(/)(\d+)(/?)")
             .expect("regex creation failed");
     static ref SHORT_URL_REGEX: Regex =
-        Regex::new(r"(https://flic.kr/p/)([\d\w]+)").expect("regex creation failed");
+        Regex::new(r"(https://flic\.kr/p/)(\w+)").expect("regex creation failed");
 }
 
+const ALPHABET: &str = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ";
+
 impl FlickrApiHandler {
     #[must_use]
-    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error`
+    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error
     pub fn new(info: FlickrInfo) -> Self {
         Self {
             request: ApiRequest::new(info.api_key),
         }
     }
 
-    // URL TYPE 1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
-    // URL TYPE 2: https://flic.kr/p/2oRguN3
+    // URL TYPE 1.1: https://www.flickr.com/photos/gerdavs/52310534489/, id group = 4
+    // URL TYPE 1.2: https://www.flickr.com/photos/198319418@N06/53077317043, id group = 4
+    // URL TYPE 2: https://flic.kr/p/2oRguN3, id group = 2
     // Both cases: Split with '/' and get last member (= photo_id).
-    fn determine_photo_id(mut url: &str) -> Result<&str> {
-        if !SHORT_URL_REGEX.is_match(url) && !LONG_URL_REGEX.is_match(url) {
-            return Err(ImageHosterError::FormatNotFound(format!(
+    fn determine_photo_id(url: &str) -> Result<String> {
+        if SHORT_URL_REGEX.is_match(url) {
+            let groups = SHORT_URL_REGEX
+                .captures(url)
+                .expect("url matches regex but captures could not be created");
+            groups.get(2).map(|m| m.as_str()).map_or_else(
+                || {
+                    Err(ImageHosterError::FormatNotFound(format!(
+                        "could not detect id group in '{url}'"
+                    )))
+                },
+                Self::decode,
+            )
+        } else if LONG_URL_REGEX.is_match(url) {
+            let groups = LONG_URL_REGEX
+                .captures(url)
+                .expect("url matches regex but captures could not be created");
+            groups.get(4).map(|m| m.as_str()).map_or_else(
```

Comment:
How to handle these 'magic-numbers'?

Reply:
as you only need to extract one bit out of the string, you could use avoid using `()` everywhere else. Then you would only have one capture group, you can index with 1 (0 is always the entire match string). If you then can guarantee, that such an capture group always exists (which you can with the regex provided), you can simply access it by `groups[1]` and dont have to work with an option.